common/arg.cpp

```diff
@@ -833,9 +833,11 @@ static bool common_params_parse_ex(int argc, char ** argv, common_params_context
 
     // allow --mmproj to be set from -hf
     // assuming that mmproj is always in the same repo as text model
-    if (!params.model.hf_repo.empty() && ctx_arg.ex == LLAMA_EXAMPLE_LLAVA) {
+    if (!params.model.hf_repo.empty() && (
+            ctx_arg.ex == LLAMA_EXAMPLE_LLAVA || ctx_arg.ex == LLAMA_EXAMPLE_SERVER)) {
         params.mmproj.hf_repo = params.model.hf_repo;
     }
+    // TODO @ngxson : this will break non-vision model with -hf, need to fix before merging
     common_params_handle_model(params.mmproj,            params.hf_token, "", true);
```

Comment:
@ngxson Is it possible to add a `--no-offload-mmproj` param here to keep the mmproj model on the CPU and the larger text model on GPU?

We can use mtmd_param_param: use_gpu=false to keep the projector model on CPU itself.

https://github.com/ggml-org/llama.cpp/blob/0019279bb56f028e4eee19b59b19750736c719f7/examples/llava/mtmd.h#L52-L58

It will be useful where the GPU VRAM is limited.

Reply:
It's added in https://github.com/ggml-org/llama.cpp/pull/13093

the flag is `--no-mmproj-offload` and not `--no-offload-mmproj`, to align with the existing `--no-kv-offload`

Reply:
Thanks alot!!!